web/frontend/src/components/base.vue

```diff
@@ -78,24 +84,66 @@ const setDirection = (dir: Directions) => {
   direction.value = dir;
 };
 
-const handleBaseActionStop = (name: string) => {
+const stop = () => {
   const req = new baseApi.StopRequest();
-  req.setName(name);
+  req.setName(props.name);
   window.baseService.stop(req, new grpc.Metadata(), displayError);
 };
 
-const baseKeyboardCtl = (name: string, controls: Record<string, boolean>) => {
-  if (Object.values(controls).every((item) => item === false)) {
-    handleBaseActionStop(name);
-    return;
+const digestInput = () => {
+  let linearValue = 0;
+  let angularValue = 0;
+
+  for (const item of pressed) {
```

Comment:
Here we loop through the stack of pressed keys, and apply weights for each key. We don't need to worry about values growing larger than 1, because all items in a `Set` can be guaranteed to be unique.

The result of this behavior is that pressing the left and right keys (or the up and down keys) will cancel each other out.